runtime/hsa-runtime/core/inc/driver.h

```diff
@@ -63,14 +64,37 @@ enum class DriverQuery { GET_DRIVER_VERSION };
 
 enum class DriverType { XDNA = 0, KFD, NUM_DRIVER_TYPES };
 
+/// @brief Handle for exported / imported memory.
+struct ShareableHandle {
+  uint64_t handle{};
+
+  bool IsValid() const { return handle != 0; }
+};
+
 /// @brief Kernel driver interface.
 ///
 /// @details A class used to provide an interface between the core runtime
 /// and agent kernel drivers. It also maintains state associated with active
 /// kernel drivers.
 class Driver {
- public:
-  Driver() = delete;
+public:
+  /// @brief Converts @ref hsa_access_permission_t to mmap memory protection
+  ///        flags.
+  __forceinline static int
+  PermissionsToMmapFlags(hsa_access_permission_t perms) {
```

Comment:
The core::Driver should be reusable across Windows/Linux so should we move this to the utils OS lib? Or at the very least guard the Linux specific code here with an `#ifdef __linux__`?

Reply:
Moved to its own header file. There wasn't one for memory, so I added it.